packages/veraswap-app/src/atoms/atoms.ts

```diff
@@ -98,9 +98,15 @@ export const swapStepAtom = atom((get) => {
         tokenInAmount > tokenInBridgeAllowance
     ) {
         return SwapStep.APPROVE_BRIDGE;
-    } else if (tokenInPermit2Allowance === null || tokenInPermit2Allowance < tokenInAmount) {
+    } else if (
+        tokenIn.standard !== "NativeToken" &&
+        (tokenInPermit2Allowance === null || tokenInPermit2Allowance < tokenInAmount)
+    ) {
         return SwapStep.APPROVE_PERMIT2;
-    } else if (tokenInUniswapRouterAllowance === null || tokenInUniswapRouterAllowance < tokenInAmount) {
+    } else if (
+        transactionType?.type !== "BRIDGE" &&
```

Comment:
- Current implementation is incorrect because still requires PERMIT2 when doing `BRIDGE_SWAP`
- Better check the affirmative `type == SWAP || type == SWAP_BRIDGE`
- Ideally in future APPROVE_PERMIT2 just requests a signature (when needed)